core/util/util.go

```diff
@@ -399,14 +399,14 @@ func jsonPath(query, toMatch string) interface{} {
 		return ""
 	}
 
-	// Jsonpath library converts large int into a string with scientific notion, the following
-	// reverts that process to avoid mismatching when using the jsonpath result for csv data lookup
-	floatResult, err := strconv.ParseFloat(result, 64)
-	// if the string is a float and a whole number
-	if err == nil && floatResult == float64(int64(floatResult)) {
-		intResult := int(floatResult)
-		result = strconv.Itoa(intResult)
-	}
+	//// Jsonpath library converts large int into a string with scientific notion, the following
+	//// reverts that process to avoid mismatching when using the jsonpath result for csv data lookup
+	//floatResult, err := strconv.ParseFloat(result, 64)
+	//// if the string is a float and a whole number
+	//if err == nil && floatResult == float64(int64(floatResult)) {
+	//	intResult := int(floatResult)
+	//	result = strconv.Itoa(intResult)
+	//}
```

Comment:
but this introduces another issue...

Reply:
This is not correct way to do things. I am not sure why we added this hack.

Reply:
so what would be the correct way?

Reply:
@tommysitu  is that change still needed?

Reply:
yes, it's needed. But commenting it out, several tests will fail, eg. `Test_ApplyTemplate_PutAndGetValue`

Reply:
@tommysitu I have added handling for it.